.github/workflows/pull-request-checks.yml

```diff
@@ -0,0 +1,91 @@
+name: Pull Request Checks
+on:
+    pull_request:
+        branches: [main]
+env:
+    NPM_AUTH: ${{ secrets.ARTIFACTORY_NPM_AUTH }}
+    NPM_EMAIL: ${{ secrets.ARTIFACTORY_NPM_EMAIL }}
+jobs:
+    check-formatting-and-lint:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout target branch
+              uses: actions/checkout@v2
+            - name: Use Node 16.x
+              uses: actions/setup-node@v2
+              with:
+                  node-version: '16.x'
+            - name: Cache node modules
+              uses: actions/cache@v2
+              id: cache
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-node-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-node-
+            - name: Install dependencies
+              if: steps.cache.outputs.cache-hit != 'true'
+              run: yarn install --frozen-lockfile
+            - name: Run Prettier
+              run: yarn format:check
+            - name: Lint
+              run: yarn lint
+    cypress-tests:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout target branch
+              uses: actions/checkout@v2
+            - name: Use Node 16.x
+              uses: actions/setup-node@v2
+              with:
+                  node-version: '16.x'
+            - name: Cache node modules
+              uses: actions/cache@v2
+              id: cache
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-node-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-node-
+            - name: Install dependencies
+              if: steps.cache.outputs.cache-hit != 'true'
+              run: yarn install --frozen-lockfile
+            - name: Install Cypress
+              if: steps.cache.outputs.cache-hit == 'true'
+              run: ./node_modules/.bin/cypress install
+            - name: Build
+              run: yarn build
```

Comment:
So I think `yarn start` will just start the server in production mode, but it needs to be built before that with `yarn build`. Here's the Next.js docs for the start command https://nextjs.org/docs/api-reference/cli#production.

Reply:
Oh I agree we need yarn build, but may be we will have another step called Deploy for that at a later point. Only wondering if yarn build is needed in Cypress Run step.

Reply:
I think so since Cypress needs the app to be built to test against it, and since all of the jobs in this step run at the same time on different machines, we do need to build it for the cypress job.

Reply:
👍 . I'm good then. LGTM.